packages/xo-web/src/common/editable/index.js

```diff
@@ -343,70 +337,35 @@ export class Number extends Component {
 
 export class Select extends Editable {
   static propTypes = {
-    options: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
-      .isRequired,
     renderer: PropTypes.func,
-  }
-
-  componentWillReceiveProps(props) {
-    if (
-      props.value !== this.props.value ||
-      props.options !== this.props.options
-    ) {
-      this.setState({
-        valueKey: findKey(props.options, option => option === props.value),
-      })
-    }
+    value: PropTypes.oneOfType([PropTypes.oneOf([null]), PropTypes.object]),
   }
 
   get value() {
-    return this.props.options[this.state.valueKey]
-  }
-
-  _onChange = event => {
-    this.setState({ valueKey: getEventValue(event) }, this._save)
-  }
-
-  _optionToJsx = (option, key) => {
-    const { renderer } = this.props
-
-    return (
-      <option key={key} value={key}>
-        {renderer ? renderer(option) : option}
-      </option>
-    )
+    return this.state.value
   }
 
-  _onEditionMount = ref => {
-    // Seems to work in Google Chrome (not in Firefox)
-    ref && ref.dispatchEvent(new window.MouseEvent('mousedown'))
-  }
+  _onChange = value => this.setState({ value }, value && this._save)
 
   _renderDisplay() {
     const { children, renderer, value } = this.props
-
-    return children || <span>{renderer ? renderer(value) : value}</span>
-  }
-
-  _renderEdition() {
-    const { saving, valueKey } = this.state
-    const { options } = this.props
-
     return (
-      <select
-        autoFocus
-        className={classNames('form-control', styles.select)}
-        onBlur={this._closeEdition}
-        onChange={this._onChange}
-        onKeyDown={this._onKeyDown}
-        readOnly={saving}
-        ref={this._onEditionMount}
-        value={valueKey}
-      >
-        {map(options, this._optionToJsx)}
-      </select>
+      children || (
+        <span>{renderer !== undefined ? renderer(value) : value}</span>
+      )
     )
   }
+
+  _renderEdition = () => (
+    <FormSelect
+      {...this.props}
+      autoFocus
+      onBlur={this._closeEdition}
+      onChange={this._onChange}
+      onKeyDown={this._onKeyDown}
```

Comment:
So it will `_save` when we press Enter?

Reply:
yes

Reply:
What if we want to select the values with the keyboard with `multi`?